src/Components/TimeSheets/Form/timesheetForm.js

```diff
@@ -76,53 +83,29 @@ const Form = () => {
 
   const onSubmit = async (event) => {
     if (!isEditing) {
-      event.preventDefault();
-      const response = await fetch(`${process.env.REACT_APP_API_URL}/time-sheets`, {
-        method: 'POST',
-        headers: { 'Content-Type': 'application/json' },
-        body: JSON.stringify({
-          description: timesheetAdded.description,
-          date: timesheetAdded.date,
-          hours: timesheetAdded.hours,
-          project: timesheetAdded.project,
-          employee: timesheetAdded.employee,
-          task: timesheetAdded.task
-        })
-      });
-      const data = await response.json();
-      if (!data.error) {
+      const response = await dispatch(createTimesheet(timesheetAdded));
+      if (response.type === POST_TIMESHEETS_SUCCESS) {
         history.push('/time-sheets');
-      } else {
-        setErrorState(data.message);
       }
     } else {
       const id = params.id;
       event.preventDefault();
-      const response = await fetch(`${process.env.REACT_APP_API_URL}/time-sheets/${id}`, {
-        method: 'PUT',
-        headers: { 'Content-Type': 'application/json' },
-        body: JSON.stringify({
-          description: timesheetAdded.description,
-          date: timesheetAdded.date,
-          hours: timesheetAdded.hours,
-          project: timesheetAdded.project,
-          employee: timesheetAdded.employee,
-          task: timesheetAdded.task
-        })
-      });
-      const data = await response.json();
-      if (!data.error) {
+      const response = await dispatch(editTimesheet(id, timesheetAdded));
+      if (response.type === PUT_TIMESHEET_SUCCESS) {
         history.push('/time-sheets');
-      } else {
-        setErrorState(data.message);
       }
     }
   };
+
+  if (isLoading) {
```

Comment:
We are rendering this only when we are fetching the timesheets, but we should render this loading also when we are fetching the projects and the tasks. So we should have something like:
```suggestion
  if (isLoadingTimesheets || isLoadingTasks || isLoadingProjects) {
```

each isLoading comes from redux state.